modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/largecluster/LargeClusterFactory.java

```diff
@@ -77,8 +77,11 @@ public ExecutionTarget partitioned(List<TokenizedAssignments> assignments) {
             BitSet nodes = new BitSet(assignment.nodes().size());
             for (Assignment a : assignment.nodes()) {
                 int node = nodeNameToId.getOrDefault(a.consistentId(), -1);
-                assert node >= 0 : "invalid node";
-                nodes.set(node);
+
+                // TODO Ignore unknown node until IGNITE-22969
+                if (node != -1) {
+                    nodes.set(node);
+                }
             }
 
             partitionNodes[idx] = nodes;
```

Comment:
let's add validation, that `nodes` set contains at least one node. We can't just ignore partition if all hosting nodes unavailable. Besides, we need to extend tests to cover both cases: partition partially available and partition is not available at all.